block/base/handlers.go

```diff
@@ -170,3 +170,19 @@ func DefaultMatchHandler() MatchHandler {
 		return true
 	}
 }
+
+// NewMatchHandler returns a match handler that matches transactions
+// that match the lane and do not match with any of the provided match handlers.
+// In the context of building an application, you would want to use this to
+// ignore the match handlers of other lanes in the application.
+func NewMatchHandler(mh MatchHandler, ignoreMHs ...MatchHandler) MatchHandler {
+	return func(ctx sdk.Context, tx sdk.Tx) bool {
```

Comment:
hmm... so we're still forcing the devs to pass in the other MatchHandlers to ignore? IMO this is still a foot-gun, as for any other lanes that they're importing they need to be aware of the implementations / what txs they match (otherwise you risk the potential of a deadlock)

Reply:
Is it possible to have some validation on the set of configured MatchHandlers to determine if there are any potential deadlocks?

Reply:
tbh this is redudant code for this particular example, so ill remove the it from set up in lanes.go

Reply:
I think the option for `NetMatchHandler` is still useful in case people want, ill just updates docs for that now.